jOOQ/src/main/java/org/jooq/tools/jdbc/JDBCUtils.java

```diff
@@ -119,6 +129,66 @@ public static final SQLDialect dialect(Connection connection) {
         return result;
     }
 
+    public static final SQLDialect dialect(String url, int majorVersion, int minorVersion) {
+        SQLDialect dialect = dialect( url );
+
+        switch ( dialect ) {
+            case POSTGRES:
+                return getPostgreSqlDialect( majorVersion, minorVersion );
+            case MYSQL:
+                return getMySqlDialect( majorVersion, minorVersion );
+            case FIREBIRD:
+                return getFirebirdDialect( majorVersion, minorVersion );
+        }
+
+        return dialect;
+    }
+
+    public static final SQLDialect getPostgreSqlDialect(int majorVersion, int minorVersion) {
```

Comment:
I'd prefer these to be private until someone can show a use-case why they should be public...